### ISSUE TITLE
Handle some proxying header?

### ISSUE THREAD

**Issue by nabijaczleweli** [thecoshman/http#98]
A common one is `X-Forwarded-For`, and it'd be useful for running this behind a reverse proxy like `nginx` (like I do, on, e.g. https://foreign.nabijaczleweli.xyz/debian); right now my journal looks like this:

```
Feb 15 19:47:51 tarta http[5689]: [2020-02-15 19:47:51] 127.0.0.1:54280 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:11:46 tarta http[5689]: [2020-02-15 20:11:46] 127.0.0.1:54448 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:14:23 tarta http[5689]: [2020-02-15 20:14:23] 127.0.0.1:54474 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/dists/buster/InRelease as text/plain
Feb 15 20:14:23 tarta http[5689]: [2020-02-15 20:14:23] 127.0.0.1:54476 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/dists/sid/InRelease as text/plain
Feb 15 20:14:23 tarta http[5689]: [2020-02-15 20:14:23] 127.0.0.1:54480 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/dists/buster/main/source/Sources.gz as application/x-gzip
Feb 15 20:14:23 tarta http[5689]: [2020-02-15 20:14:23] 127.0.0.1:54482 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/dists/buster/main/binary-amd64/Packages.gz as application/x-gzip
Feb 15 20:14:40 tarta http[5689]: [2020-02-15 20:14:40] 127.0.0.1:54492 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/pool/main/h/http/http_1.8.0-1_amd64.deb as application/octet-stream
Feb 15 20:16:26 tarta http[5689]: [2020-02-15 20:16:26] 127.0.0.1:54506 was served directory listing for /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian
Feb 15 20:16:26 tarta http[5689]: [2020-02-15 20:16:26]                 encoded as gzip for 345.6% ratio
Feb 15 20:16:30 tarta http[5689]: [2020-02-15 20:16:30] 127.0.0.1:54512 was served directory listing for /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/pool
Feb 15 20:16:30 tarta http[5689]: [2020-02-15 20:16:30]                 encoded as gzip for 338.1% ratio
Feb 15 20:16:31 tarta http[5689]: [2020-02-15 20:16:31] 127.0.0.1:54514 was served directory listing for /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/pool/patches
Feb 15 20:16:31 tarta http[5689]: [2020-02-15 20:16:31]                 encoded as gzip for 338.7% ratio
Feb 15 20:16:32 tarta http[5689]: [2020-02-15 20:16:32] 127.0.0.1:54516 was served directory listing for /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/pool/patches/h
Feb 15 20:16:32 tarta http[5689]: [2020-02-15 20:16:32]                 encoded as gzip for 337.8% ratio
Feb 15 20:16:34 tarta http[5689]: [2020-02-15 20:16:34] 127.0.0.1:54518 was served directory listing for /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/pool/patches/h/http
Feb 15 20:16:34 tarta http[5689]: [2020-02-15 20:16:34]                 encoded as gzip for 338.2% ratio
Feb 15 20:21:37 tarta http[5689]: [2020-02-15 20:21:37] 127.0.0.1:54552 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:25:28 tarta http[5689]: [2020-02-15 20:25:28] 127.0.0.1:54574 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:27:53 tarta http[5689]: [2020-02-15 20:27:53] 127.0.0.1:54600 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:36:38 tarta http[5689]: [2020-02-15 20:36:38] 127.0.0.1:54662 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:36:42 tarta http[5689]: [2020-02-15 20:36:42] 127.0.0.1:54664 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
Feb 15 20:37:06 tarta http[5689]: [2020-02-15 20:37:06] 127.0.0.1:54666 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
```

Where, and of this I'm quite sure, *none* of the requests originated from the server.

I propose using the value of `X-Forwarded-For`, if present, instead of the source address, with some marker, like, e.g.
```
Feb 15 20:21:37 tarta http[5689]: [2020-02-15 20:21:37] 127.0.0.1:54552 (for 192.168.1.167:6969) was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
```
or
```
Feb 15 20:21:37 tarta http[5689]: [2020-02-15 20:21:37] 127.0.0.1:54552 for 192.168.1.167:6969 was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
```
or
```
Feb 15 20:21:37 tarta http[5689]: [2020-02-15 20:21:37] 127.0.0.1:54552 [for 192.168.1.167:6969] was served file /home/nabijaczleweli/foreign.nabijaczleweli.xyz/debian/README as text/plain
```

**Comment by nabijaczleweli** [comment]
> [21:40] Cat Plus Plus: If you want to use X-Forwarded-For then remember to have a strict whitelist of host you trust to set it

Which is a good point, I propose a multivalue `--allow-proxy-from` arg that takes a list of IPs for which to handle the header (and ignore it otherwise).

**Comment by nabijaczleweli** [comment]
Implementation yielding this:
```
p:\Rust\http>target\debug\http --proxy X-Forwarded-For:127.0.0.1 --proxy X-Proxied-For:192.168.1.0/24
Hosting "." on port 8000 without TLS and no authentication...
Trusted proxies:
Header           Network
X-Forwarded-For  127.0.0.1
X-Proxied-For    192.168.1.0/24
Ctrl-C to stop.

[2020-02-17 17:48:41] 127.0.0.1:1392 for OwO was served directory listing for \\?\P:\Rust\http
[2020-02-17 17:49:12] 192.168.1.109:1403 for UwU was served directory listing for \\?\P:\Rust\http
[2020-02-17 17:49:29] 93.184.216.34:1397 was served directory listing for \\?\P:\Rust\http
```
in referenced commits.

**Comment by nabijaczleweli** [comment]
Released in [`v1.9.0`](https://github.com/thecoshman/http/releases/tag/v1.9.0)
